spock-core/src/main/java/org/spockframework/runtime/model/MethodInfo.java

```diff
@@ -122,12 +124,23 @@
   public Object invoke(Object target, Object... arguments) throws Throwable {
     for (int i = 0, argCount = arguments.length; i < argCount; i++) {
       if (arguments[i] == MISSING_ARGUMENT) {
-        StringJoiner missingArguments = new StringJoiner("', '", "No argument was provided for parameters: '", "'");
-        List<String> parameterNames = getFeature().getParameterNames();
-        missingArguments.add(parameterNames.get(i));
-        for (int j = i + 1; j < argCount; j++) {
-          if (arguments[j] == MISSING_ARGUMENT) {
-            missingArguments.add(parameterNames.get(j));
+        StringJoiner missingArguments;
+        if (getKind() == FEATURE) {
+          missingArguments = new StringJoiner("', '", "No argument was provided for parameters: '", "'");
+          List<String> parameterNames = getFeature().getParameterNames();
+          missingArguments.add(parameterNames.get(i));
+          for (int j = i + 1; j < argCount; j++) {
+            if (arguments[j] == MISSING_ARGUMENT) {
+              missingArguments.add(parameterNames.get(j));
+            }
+          }
+        } else {
+          missingArguments = new StringJoiner(", ", "No argument was provided for parameters: ", "");
+          missingArguments.add(String.valueOf(i));
```

Comment:
I would use `getReflection().getParameters()[i].getName()` it eithers gives `arg<i>` or the real name when compiled with `-parameters`, would need to check for Groovy's behavior

Reply:
Good point